prow/cmd/branchprotector/README.md

```diff
@@ -0,0 +1,170 @@
+# Branch Protection Documentation
+
+branchprotector configures [github branch protection] according to a specified
+policy.
+
+## Policy configuration
+
+Extend the primary prow [`config.yaml`] document to include a top-level
+`branch-protection` key that looks like the following:
+
+```yaml
+
+branch-protection:
+  orgs:
+    kubernetes:
+      repos:
+        test-infra:
+          # Protect all branches in kubernetes/test-infra
+          protect-by-default: true
+          # Always allow the org's oncall-team to push
+          pushers: ["oncall-team"]
+          # Ensure that the extra-process-followed github status context passes.
+          # In addition, adds any always_run: True prow jobs
+          require-contexts: ["extra-process-followed"]
+
+presubmits:
+  kubernetes/test-infra:
+  - name: fancy-job-name
+    context: fancy-job-name
+    always_run: true
+    spec:  # podspec that runs job
+```
+
+This config will:
+  * Enable protection for every branch in the `kubernetes/test-infra`
+repo.
+  * Require `extra-process-followed` and `fancy-job-name` [status contexts] to pass
+    before allowing a merge
+    - Although it will always allow `oncall-team` to merge, even if required
+      contexts fail.
+    - Note that `fancy-job-name` is pulled in automatically from the
+      `presubmits` config for the repo, if one exists.
+
+### Updating
+
+* Send PR with `config.yaml` changes
+* Merge PR
+* Done!
+
+Make changes to the policy by modifying [`config.yaml`] in your favorite text
+editor and then send out a PR. When the PR merges prow pushes the updated config
+. The next time the branch protection job runs it will apply the new policies.
```

Comment:
Maybe mention the run frequency so that users know how long they should expect to wait for changes to be applied.

Reply:
Done